core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -1039,8 +1043,9 @@ object IO extends Serializable {
    */
   final def unsandbox[E, A](v: IO[Either[List[Throwable], E], A]): IO[E, A] =
     v.catchAll[E, A] {
-      case Right(e) => IO.fail(e)
-      case Left(ts) => IO.terminate0(ts)
+      case Right(e)      => IO.fail(e)
```

Comment:
How about catchAll? Still taking an `E`?

Reply:
Isn't this going to be
```scala
def sandbox[E1 <: E]: IO[Cause[E1], A]
```
? If the action failed because of an unchecked error or interruption, we get a `Cause[Nothing]`.